internal/generate/servicepackage/file.gtpl

```diff
@@ -236,6 +239,18 @@ func (p *servicePackage) NewClient(ctx context.Context, config map[string]any) (
 				o.Region = region
 			}
 		},
+		func(o *{{ .GoV2Package }}.Options) {
+			if inContext, ok := conns.FromContext(ctx); ok && inContext.VCREnabled() {
+				tflog.Info(ctx, "overriding retry behavior to immediately return VCR errors")
+				o.Retryer = conns.AddIsErrorRetryables(cfg.Retryer().(aws.RetryerV2), retry.IsErrorRetryableFunc(func(err error) aws.Ternary {
```

Comment:
Can this anonymous function be extracted into a common package and referenced here?

Reply:
Good catch. Will move it now.